src/mx_bluesky/common/device_setup_plans/gonio.py

```diff
@@ -0,0 +1,30 @@
+import numpy as np
+from bluesky import plan_stubs as bps
+from bluesky.utils import FailedStatus
+from dodal.devices.motors import XYZOmegaStage
+from ophyd_async.epics.motor import MotorLimitsError
+
+from mx_bluesky.common.utils.exceptions import SampleError
+
+
+def move_gonio_warn_on_out_of_range(
+    gonio: XYZOmegaStage,
+    position: np.ndarray | list[float] | tuple[float, float, float],
+):
+    """
+    Throws a SampleException if the specified position is out of range for the
+    gonio. Otherwise moves to that position. The check is from ophyd-async
+    """
+    group = "move_warn_out_of_range"
+    try:
+        yield from bps.abs_set(gonio.x, position[0], group=group, wait=True)
+        yield from bps.abs_set(gonio.y, position[1], group=group, wait=True)
+        yield from bps.abs_set(gonio.z, position[2], group=group, wait=True)
+        yield from bps.wait(group=group)
```

Comment:
Should: The `wait=True` on each of these means that we're waiting for each move to finish before doing the next one. This means that the `bps.wait` at the end isn't doing anything. If you remove all the `wait=True` it will move them all at the same time. Alternatively you can do:
```suggestion
        yield from bps.mv(gonio.x, position[0], gonio.y, position[1], gonio.z, position[2])
```
To have the same behavior in fewer lines